### PR TITLE
Transip provider

### PR DESCRIPTION
**Pull request by blackshadev**
Add functionality to use TransIP DNS

 - [x] GetNameservers
 - [x]  GetZoneRecords
 - [x] GetDomainCorrections
 - [x] Added docs
 - [x] Test if it works

### CODE REVIEW

File: providers/transip/transipProvider.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
+package transip
+
```
Comment (tlimoncelli) [inline providers/transip/transipProvider.go:2]: Please add a comment with a little into about transip (i.e. the URL, the company name, etc.).

File: providers/transip/transipProvider.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	"github.com/transip/gotransip/v6/repository"
+)
+
+type transipProvider struct {
+	client  *repository.Client
+	domains *domain.Repository
+}
+
+var features = providers.DocumentationNotes{
+	providers.DocCreateDomains:       providers.Cannot(),
```
Comment (tlimoncelli) [inline providers/transip/transipProvider.go:21]: Please sort this list.

File: providers/transip/transipProvider.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+	case "SSHFP":
+		return fmt.Sprintf("%d %d %s", rc.SshfpAlgorithm, rc.SshfpFingerprint, rc.GetTargetField())
+	case "DS":
+		return fmt.Sprintf("%d %d %d %s", rc.DsKeyTag, rc.DsAlgorithm, rc.DsDigestType, rc.DsDigest)
+	case "SRV":
+		return fmt.Sprintf("%d %d %s", rc.SrvWeight, rc.SrvPort, rc.GetTargetField())
+	case "TXT":
+		quoted := make([]string, len(rc.TxtStrings))
+		for i := range rc.TxtStrings {
+			quoted[i] = quoteDNSString(rc.TxtStrings[i])
```
Comment (tlimoncelli) [inline providers/transip/transipProvider.go]: I don't see where quoteDNSString is defined.  Missing file?

**Review by tlimoncelli — COMMENTED** [review]
Looking great so far!  Very clean code, by the way!

I just had a few minor suggestions.

Please remember when you are done to make sure all documentation is up to date (Step 8) and that you've checked your work (Step 13).  https://stackexchange.github.io/dnscontrol/writing-providers

Thank you for contributing to the DNSControl project!

File: providers/transip/transipProvider.go (hunk 3 above)
Comment (blackshadev) [reply providers/transip/transipProvider.go]: I copied allot of code from other providers, I ended up not needing the quoteDNSString , so I removed it.

**Comment by blackshadev** [comment]
> Looking great so far! Very clean code, by the way!
> 
> I just had a few minor suggestions.
> 
> Please remember when you are done to make sure all documentation is up to date (Step 8) and that you've checked your work (Step 13). https://stackexchange.github.io/dnscontrol/writing-providers
> 
> Thank you for contributing to the DNSControl project!

Hi Thanks! Allot of code was copied from other providers, so I can't take all credit on that part.

I do need to test it before continue-ing. But most of the code should be fine.

**Comment by blackshadev** [comment]
This should be ready to go.

**Comment by tlimoncelli** [comment]
Please rebase

**Comment by blackshadev** [comment]
> Please rebase

Done

File: docs/_providers/transip.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+{
+  "transip":{
+    "AccessToken": "your-transip-personal-access-token"
+  }
+}
+{% endhighlight %}
+
+## Metadata
+
+This provider does not recognize any special metadata fields unique to Vultr.
```
Comment (svenluijten) [inline docs/_providers/transip.md:24]:
```suggestion
This provider does not recognize any special metadata fields unique to TransIP.
```

**Comment by tlimoncelli** [comment]
Thanks for adding a new provider to the project!  I'm always excited to see the project grow this way!

**Pull request merged**